indexer/src/configs.rs

```diff
@@ -0,0 +1,168 @@
+    }
+}
+
+impl Opts {
+    pub async fn to_lake_config(&self) -> near_lake_framework::LakeConfig {
+        let config_builder = near_lake_framework::LakeConfigBuilder::default();
+
+        match &self.chain_id {
+            ChainId::Mainnet(_) => config_builder
+                .mainnet()
+                .start_block_height(get_start_block_height(self).await),
+            ChainId::Testnet(_) => config_builder
+                .testnet()
+                .start_block_height(get_start_block_height(self).await),
+        }
+        .build()
+        .expect("Failed to build LakeConfig")
+    }
+}
+
+async fn get_start_block_height(opts: &Opts) -> u64 {
+    match opts.start_options() {
+        StartOptions::FromBlock { height } => *height,
+        StartOptions::FromInterruption => {
+            let pool = models::establish_connection(&opts.database_url);
+            let last_indexed_block: u64 = match adapters::blocks::latest_block_height(&pool).await {
+                Ok(last_indexed_block) => {
+                    if let Some(last_indexed_block) = last_indexed_block {
+                        last_indexed_block
+                    } else {
+                        final_block_height(opts).await
+                    }
+                }
+                Err(err) => {
+                    tracing::warn!(
+                        target: "alertexer",
+                        "Failed to get last indexer block from Database. Failing to the latest one...\n{:#?}",
+                        err
+                    );
+                    final_block_height(opts).await
+                }
+            };
+            last_indexed_block
+        }
+        StartOptions::FromLatest => final_block_height(opts).await,
+    }
+}
+
+pub(crate) fn init_tracing(debug: bool) -> anyhow::Result<()> {
+    let mut env_filter =
+        EnvFilter::new("near_lake_framework=info,indexer_for_explorer=info,stats=info");
+
+    if debug {
+        env_filter = env_filter
+            .add_directive("indexer_for_explorer=debug".parse()?)
+            .add_directive("near_lake_framework=debug".parse()?);
+    }
+
+    if let Ok(rust_log) = std::env::var("RUST_LOG") {
+        if !rust_log.is_empty() {
+            for directive in rust_log.split(',').filter_map(|s| match s.parse() {
+                Ok(directive) => Some(directive),
+                Err(err) => {
+                    eprintln!("Ignoring directive `{}`: {}", s, err);
+                    None
+                }
+            }) {
+                env_filter = env_filter.add_directive(directive);
+            }
+        }
+    }
+
+    let subscriber = tracing_subscriber::fmt::Subscriber::builder()
+        .with_env_filter(env_filter)
+        .with_writer(std::io::stderr);
+
+    if std::env::var("ENABLE_JSON_LOGS").is_ok() {
+        subscriber.json().init();
+    } else {
+        subscriber.compact().init();
+    }
+
+    Ok(())
+}
+
+async fn final_block_height(opts: &Opts) -> u64 {
```

Comment:
It would be nice if the parameters help the reader to understand the logic without reading it.
Let's create rpc client only at main and pass it everywhere where it's needed? We almost there, it's the only exception

Reply:
I would suggest extracting this proposition to a separate issue for improvement. It is out of the scope of this PR and shouldn't block it